diff --git a/arch/arm/mach-tegra/board-colibri_t30.c b/arch/arm/mach-tegra/board-colibri_t30.c
--- a/arch/arm/mach-tegra/board-colibri_t30.c
+++ b/arch/arm/mach-tegra/board-colibri_t30.c
@@ -42,6 +42,9 @@
 #include <media/soc_camera.h>
 #include <media/tegra_v4l2_camera.h>
 
+ #include <linux/input.h>
+ #include <linux/rotary_encoder.h>
+
 #include "board-colibri_t30.h"
 #include "board.h"
 #include "clock.h"
@@ -542,8 +545,8 @@ static struct gpio colibri_t30_gpios[] = {
 //multiplexed LCD_D18
 //	{TEGRA_GPIO_PX3,	GPIOF_IN,	"SODIMM pin 136"},
 	{TEGRA_GPIO_PX4,	GPIOF_IN,	"SODIMM pin 134"},
-	{TEGRA_GPIO_PX6,	GPIOF_IN,	"102, I X13 ForceOFF#"},
-	{TEGRA_GPIO_PX7,	GPIOF_IN,	"104, I X14 ForceOFF#"},
+//	{TEGRA_GPIO_PX6,	GPIOF_IN,	"102, I X13 ForceOFF#"},
+//	{TEGRA_GPIO_PX7,	GPIOF_IN,	"104, I X14 ForceOFF#"},
 #ifndef COLIBRI_T30_VI
 //	{TEGRA_GPIO_PY4,	GPIOF_IN,	"SODI-97, Iris X16-17"},
 //	{TEGRA_GPIO_PY5,	GPIOF_IN,	"SODI-79, Iris X16-19"},
@@ -575,6 +578,104 @@ static struct gpio colibri_t30_gpios[] = {
 
 };
 
+/* Rotary encoder */
+#define GPIO_ROTARY0_A TEGRA_GPIO_PK4
+#define GPIO_ROTARY0_B TEGRA_GPIO_PX7
+#define GPIO_ROTARY1_A TEGRA_GPIO_PX6
+#ifdef TEST_ON_EVAL_BOARD // todo -- where to set this variable ???
+#define GPIO_ROTARY1_B TEGRA_GPIO_PCC2
+#else
+#define GPIO_ROTARY1_B TEGRA_GPIO_PCC2 ////// #define GPIO_ROTARY1_B TEGRA_GPIO_PZ1
+#endif
+static struct rotary_encoder_platform_data my_rotary_encoder_info0 = {
+        .steps         = 15,
+        .axis          = REL_X,
+        .relative_axis = true,
+        .rollover      = true,
+        .gpio_a                = GPIO_ROTARY0_A,
+        .gpio_b                = GPIO_ROTARY0_B,
+        .inverted_a    = 0,
+        .inverted_b    = 0,
+        .half_period   = false,
+};
+
+static struct platform_device rotary_encoder_device0 = {
+        .name          = "rotary-encoder",
+        .id            = 0,
+        .dev           = {
+                .platform_data = &my_rotary_encoder_info0,
+        }
+};
+
+static struct rotary_encoder_platform_data my_rotary_encoder_info1 = {
+        .steps         = 15,
+        .axis          = REL_Y,
+        .relative_axis = true,
+        .rollover      = true,
+        .gpio_a                = GPIO_ROTARY1_A,
+        .gpio_b                = GPIO_ROTARY1_B,
+        .inverted_a    = 0,
+        .inverted_b    = 0,
+        .half_period   = false,
+};
+
+static struct platform_device rotary_encoder_device1 = {
+        .name          = "rotary-encoder",
+        .id            = 1,
+        .dev           = {
+                .platform_data = &my_rotary_encoder_info1,
+        }
+};
+
+/* GPIO buttons */
+#define GPIO_KEY_F1  TEGRA_GPIO_PK7
+#define GPIO_KEY_F2  TEGRA_GPIO_PK2
+#define GPIO_KEY_ESC TEGRA_GPIO_PB6
+
+static struct gpio_keys_button gpio_keys_button[] = {
+        {
+                .code			= KEY_F1,
+                .type			= EV_KEY,
+                .gpio			= GPIO_KEY_F1,
+                .active_low		= 1,
+                .wakeup			= 0,
+                .debounce_interval	= 5, /* ms */
+                .desc			= "Button F1",
+        },
+        {
+                .code			= KEY_F2,
+                .type			= EV_KEY,
+                .gpio			= GPIO_KEY_F2,
+                .active_low		= 1,
+                .wakeup			= 0,
+                .debounce_interval	= 5, /* ms */
+                .desc			= "Button F2",
+        },
+        {
+                .code			= KEY_ESC,
+                .type			= EV_KEY,
+                .gpio			= GPIO_KEY_ESC,
+                .active_low		= 1,
+                .wakeup			= 0,
+                .debounce_interval	= 5, /* ms */
+                .desc			= "Button ESC",
+        },
+};
+
+static struct gpio_keys_platform_data gpio_keys_platform_data = {
+        .buttons	= gpio_keys_button,
+        .nbuttons	= ARRAY_SIZE(gpio_keys_button),
+        .rep		= 0,
+};
+
+static struct platform_device displayl_gpio_keys_device = {
+        .name	= "gpio-keys",
+        .id	= -1,
+        .dev 	= {
+                .platform_data	= &gpio_keys_platform_data,
+        }
+};
+
 static void colibri_t30_gpio_init(void)
 {
 	int i = 0;
@@ -620,7 +721,7 @@ static struct mxt_platform_data colibri_atmel_pdata = {
 #ifdef USE_CAPACITIVE_TOUCH_ADAPTER
 	.gpio_reset = TEGRA_GPIO_PA6,
 #else
-	.gpio_reset = TEGRA_GPIO_PK4,
+//	.gpio_reset = TEGRA_GPIO_PK4,
 #endif
 };
 
@@ -1718,6 +1819,9 @@ static struct platform_device *colibri_t30_devices[] __initdata = {
 #ifdef CONFIG_W1_MASTER_TEGRA
 	&tegra_w1_device,
 #endif
+    &rotary_encoder_device0,
+    &rotary_encoder_device1,
+    &displayl_gpio_keys_device,
 };
 
 static void __init colibri_t30_init(void)
@@ -1818,6 +1922,7 @@ static const char *colibri_t30_dt_board_compat[] = {
 	NULL
 };
 
+
 #ifdef CONFIG_ANDROID
 MACHINE_START(COLIBRI_T30, "cardhu")
 #else
