diff --git a/arch/arm/mach-tegra/gpio-names.h b/arch/arm/mach-tegra/gpio-names.h
--- a/arch/arm/mach-tegra/gpio-names.h
+++ b/arch/arm/mach-tegra/gpio-names.h
@@ -164,7 +164,7 @@
 #define TEGRA_GPIO_PR5		141
 #define TEGRA_GPIO_PR6		142
 #define TEGRA_GPIO_PR7		143
-#define TEGRA_GPIO_PS0		144
+#define TEGRA_GPIO_PS0		181
 #define TEGRA_GPIO_PS1		145
 #define TEGRA_GPIO_PS2		146
 #define TEGRA_GPIO_PS3		147
