diff --git a/arch/arm/mach-tegra/board-colibri_t30.c b/arch/arm/mach-tegra/board-colibri_t30.c
--- a/arch/arm/mach-tegra/board-colibri_t30.c
+++ b/arch/arm/mach-tegra/board-colibri_t30.c
@@ -355,7 +355,7 @@
 /* Colibri EvalBoard V3.1a */

 /* active low interrupt (MCP2515 nINT) */
-#define CAN_INTERRUPT_GPIO	TEGRA_GPIO_PS0
+#define CAN_INTERRUPT_GPIO	TEGRA_GPIO_PW5

 static int __init colibri_t20_mcp2515_setup(struct spi_device *spi)
 {
