diff --git a/src/Hardware/DisplayDPI.hpp b/src/Hardware/DisplayDPI.hpp
--- a/src/Hardware/DisplayDPI.hpp
+++ b/src/Hardware/DisplayDPI.hpp
@@ -26,7 +26,7 @@

 #include "Util/Compiler.h"

-#if defined(USE_FB) || defined(MESA_KMS) || defined(ANDROID)
+#if defined(MESA_KMS) || defined(ANDROID)
 #define HAVE_DPI_DETECTION
 #endif


diff --git a/src/Screen/FB/TopCanvas.cpp b/src/Screen/FB/TopCanvas.cpp
--- a/src/Screen/FB/TopCanvas.cpp
+++ b/src/Screen/FB/TopCanvas.cpp
@@ -226,9 +226,10 @@
   };
 #endif

+#ifdef HAVE_DPI_DETECTION
   new_size = ::GetSize(vinfo);
-
   Display::ProvideSizeMM(new_size.cx, new_size.cy, vinfo.width, vinfo.height);
+#endif

 #elif defined(USE_VFB)
   /* allocate buffer as requested by caller */
