diff --git a/arch/arm/mach-tegra/board-colibri_t20.c b/arch/arm/mach-tegra/board-colibri_t20.c
--- a/arch/arm/mach-tegra/board-colibri_t20.c
+++ b/arch/arm/mach-tegra/board-colibri_t20.c
@@ -823,19 +823,11 @@
 			.type = "isl12022",
 	},
 	{
-		/* TouchRevolution Fusion 7 and 10 multi-touch controller */
-		I2C_BOARD_INFO("fusion_F0710A", 0x10),
-		.platform_data = &colibri_fusion_pdata,
-	},
-	{
 		/* Atmel MAX TS 7 multi-touch controller */
-		I2C_BOARD_INFO("atmel_mxt_ts", 0x4a),
+		I2C_BOARD_INFO("atmel_mxt_ts", 0x09),
 		.platform_data = &colibri_atmel_pdata,
-#ifdef USE_CAPACITIVE_TOUCH_ADAPTER
-		.irq = TEGRA_GPIO_TO_IRQ( TEGRA_GPIO_PB5 ),
-#else
-		.irq = TEGRA_GPIO_TO_IRQ( TEGRA_GPIO_PK3 ),
-#endif
+		.irq = TEGRA_GPIO_TO_IRQ( TEGRA_GPIO_PY7 ),
+
 	},
 };

