diff --git a/arch/arm/mach-tegra/board-colibri_t20.c b/arch/arm/mach-tegra/board-colibri_t20.c
--- a/arch/arm/mach-tegra/board-colibri_t20.c
+++ b/arch/arm/mach-tegra/board-colibri_t20.c
@@ -791,11 +791,11 @@ static void colibri_t20_gpio_init(void)
  * gpio_request muxes the GPIO function automatically, we only have to make
  * sure input/output muxing is done here...
  */
-static struct fusion_f0710a_init_data colibri_fusion_pdata = {
-	.pinmux_fusion_pins = NULL,
-	.gpio_int = TEGRA_GPIO_PB5, 	/* SO-DIMM 28: Pen down interrupt */
-	.gpio_reset = TEGRA_GPIO_PA6,	/* SO-DIMM 30: Reset interrupt */
-};
+//static struct fusion_f0710a_init_data colibri_fusion_pdata = {
+//	.pinmux_fusion_pins = NULL,
+//	.gpio_int = TEGRA_GPIO_PB5, 	/* SO-DIMM 28: Pen down interrupt */
+//	.gpio_reset = TEGRA_GPIO_PA6,	/* SO-DIMM 30: Reset interrupt */
+//};
 
 /*
  * Atmel touch screen GPIOs (using Toradex display/touch adapter)
