diff --git a/include/configs/colibri_t20.h b/include/configs/colibri_t20.h
--- a/include/configs/colibri_t20.h
+++ b/include/configs/colibri_t20.h
@@ -167,7 +167,7 @@
 		"load ${interface} ${drive}:1 ${loadaddr} flash_blk.img && " \
 		"source ${loadaddr}\0" \
 	"setup=setenv setupargs asix_mac=${ethaddr} " \
-		"consoleblank=0 no_console_suspend=1 console=tty1 " \
+		"consoleblank=0 no_console_suspend=1 vt.global_cursor_default=0 " \
 		"console=${console},${baudrate}n8 debug_uartport=lsport,3 " \
 		"${memargs}\0" \
 	"setupdate=run setsdupdate || run setusbupdate || run setethupdate\0" \


diff --git a/configs/colibri_t20_defconfig b/configs/colibri_t20_defconfig
--- a/configs/colibri_t20_defconfig
+++ b/configs/colibri_t20_defconfig
@@ -62,7 +62,7 @@
 CONFIG_G_DNL_VENDOR_NUM=0x1b67
 CONFIG_G_DNL_PRODUCT_NUM=0x4000
 CONFIG_DM_VIDEO=y
-CONFIG_VIDEO_TEGRA20=y
+# CONFIG_VIDEO_TEGRA20=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_FDT_FIXUP_PARTITIONS=y
